app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -81,12 +88,24 @@ class FirebaseAuthRepository {
           }
         }
       } catch (e: Exception) {
-        Log.d("SignInButton", "Login error: ${e.message}")
-        onErrorAction(e)
+        // This check is necessary bc the type of exception is always UndeclaredThrowableException
+        if (e.cause is NoCredentialException) {
```

Comment:
Fix: I've tried running your code, It doesn't work when no credentials are available. The issue is this line, since e.cause is a Throwable and not an Exception. Implementing the below described multiple catch clause is cleaner and should fix this issue too

Reply:
I changed it back now and it should work now, thanks!

Reply:
Thanks!